CDP4Dal/Session.cs

```diff
@@ -663,6 +676,80 @@ public async Task Read(IEnumerable<Thing> things, IQueryAttributes queryAttribut
             await this.AfterReadOrWriteOrUpdate(foundThings);
         }
 
+        /// <summary>
+        /// Reads a <see cref="CometTask" /> identified by the provided <see cref="Guid" />
+        /// </summary>
+        /// <param name="id">The <see cref="Guid"/> identifier for the <see cref="CometTask" /></param>
+        /// <exception cref="InvalidOperationException">If the <see cref="ActivePerson"/> is null, meaning that the session is not opened</exception>
+        public async Task ReadCometTask(Guid id)
+        {
+            if (this.ActivePerson == null)
+            {
+                throw new InvalidOperationException("The CometTask cannot be read when the ActivePerson is null; The Open method must be called prior to any of the Read methods");
+            }
+
+            // Create the token source
+            var cancellationTokenSource = new CancellationTokenSource();
+            var cancellationTokenKey = Guid.NewGuid();
+            this.cancellationTokenSourceDictionary.TryAdd(cancellationTokenKey, cancellationTokenSource);
+
+            try
+            {
+                this.Dal.Session = this;
+                var cometTask = await this.Dal.ReadCometTask(id, cancellationTokenSource.Token);
+                this.cometTasks[cometTask.Id] = cometTask;
+                this.CDPMessageBus.SendMessage(new CometTaskEvent(this, cometTask));

```

Comment:
should we not also do this on a write when a task is returned isntead of things? those are actually the ones we are interested in